src/test/java/com/yegor256/tojos/MnSynchronizedTest.java

```diff
@@ -0,0 +1,193 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link MnSynchronized}.
+ *
+ * @since 0.3.0
+ */
+class MnSynchronizedTest {
+
+    /**
+     * Number of threads.
+     */
+    static final int THREADS = 5;
+
+    /**
+     * The accumulator that contains a changes in the under test mono.
+     */
+    private static Collection<Collection<Map<String, String>>> changes;
+
+    /**
+     * The mono under test.
+     */
+    private Mono mono;
+
+    /**
+     * The executor.
+     */
+    private ThreadPoolExecutor executor;
+
+    /**
+     * The blocking queue.
+     */
+    private BlockingQueue<Runnable> queue;
+
+    /**
+     * The row.
+     */
+    private Map<String, String> row;
+
+    @BeforeEach
+    final void setUp() {
+        this.mono = new MnSynchronized(new MnMemory());
+        MnSynchronizedTest.changes = Collections.synchronizedList(new ArrayList<>(0));
+        this.row = Collections.synchronizedMap(new HashMap<>(0));
+        this.queue = new LinkedBlockingQueue<>(MnSynchronizedTest.THREADS);
+        this.executor = new ThreadPoolExecutor(
+            MnSynchronizedTest.THREADS,
+            MnSynchronizedTest.THREADS,
+            5L,
+            TimeUnit.SECONDS,
+            this.queue,
+            new ThreadPoolExecutor.CallerRunsPolicy()
+        );
+    }
+
+    /**
+     * Thread-safety test.
+     * In this test, we check the number of changes in MnSynchronized mono.
+     * It should be equal to the sum of the arithmetic progression over the number of threads.
+     *
+     * @throws InterruptedException When fails
+     */
+    @Test
+    final void concurrentScenario() throws InterruptedException {
+        this.executor.prestartAllCoreThreads();
+        for (int trds = 0; trds < MnSynchronizedTest.THREADS; ++trds) {
+            this.row.put(Tojos.KEY, String.valueOf(trds));
+            if (!this.queue.offer(new MnSynchronizedTest.TestTask(trds, this.mono, this.row))) {
+                throw new IllegalStateException("Can't put runnable test");
+            }
+        }
+        this.executor.shutdown();
+        assert this.executor.awaitTermination(5L, TimeUnit.SECONDS);
+        MatcherAssert.assertThat(
+            MnSynchronizedTest.totalSize(MnSynchronizedTest.changes),
+            Matchers.equalTo(MnSynchronizedTest.expectedSize())
+        );
+    }
+
+    /**
+     * The expected.
+     *
+     * @return Sum of arithmetic progression from 1 to number of threads
+     */
+    private static Integer expectedSize() {
+        final int len = MnSynchronizedTest.THREADS;
+        return len + (len - 1) * len / 2;
+    }
+
+    private static Integer totalSize(final Iterable<Collection<Map<String, String>>> accm) {
```

Comment:
Maybe we can use `this.changes` in that method? In order to remove the parameter `final Iterable<Collection<Map<String, String>>> accm`